homeassistant/components/waterfurnace.py

```diff
@@ -80,6 +86,36 @@ def __init__(self, hass, client):
         self.unit = client.unit
         self.data = None
         self._shutdown = False
+        self._fails = 0
+
+    def _reconnect(self):
```

Comment:
Do you think that this kind of logic should be moved into the waterfurnace lib as it might be useful for other users of the lib too?

Reply:
It's possible. I get a little concerned when libraries put all their own retry logic under the cover as they may not understand a larger symptom of what is going on that they should stop.

If you want I can move it down to the lower layer.

Reply:
Well if the lib would expose both, a raw connection and a managed connection, it could work.

I'm fine either way. You pick.

Reply:
Oh, apparently I never hit submit on my comments. For now I'd like to merge this and get back into the mix.